// @grafbase/sdk/dist/tests/unit/mongodb.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var index_1 = require("../../src/index");
var globals_1 = require("@jest/globals");
var utils_1 = require("../utils");
(0, globals_1.describe)('MongoDB generator', function () {
    var mongoParams = {
        name: 'Test',
        apiKey: 'SOME_KEY',
        appId: 'test',
        dataSource: 'data',
        database: 'tables'
    };
    (0, globals_1.beforeEach)(function () { return index_1.g.clear(); });
    (0, globals_1.it)('generates the minimum possible MongoDB datasource', function () {
        var mongo = index_1.connector.MongoDB(mongoParams);
        index_1.g.datasource(mongo);
        (0, globals_1.expect)((0, utils_1.renderGraphQL)((0, index_1.config)({ schema: index_1.g }))).toMatchInlineSnapshot("\n      \"extend schema\n        @mongodb(\n          name: \"Test\"\n          apiKey: \"SOME_KEY\"\n          appId: \"test\"\n          dataSource: \"data\"\n          database: \"tables\"\n        )\"\n    ");
    });
    (0, globals_1.it)('generates a simple model', function () {
        var mongo = index_1.connector.MongoDB(mongoParams);
        mongo
            .model('User', {
            id: index_1.g.id().unique().mapped('_id'),
            field: index_1.g.string()
        })
            .collection('users');
        index_1.g.datasource(mongo);
        (0, globals_1.expect)((0, utils_1.renderGraphQL)((0, index_1.config)({ schema: index_1.g }))).toMatchInlineSnapshot("\n      \"extend schema\n        @mongodb(\n          name: \"Test\"\n          apiKey: \"SOME_KEY\"\n          appId: \"test\"\n          dataSource: \"data\"\n          database: \"tables\"\n        )\n\n      type User @model(connector: \"Test\", collection: \"users\") {\n        id: ID! @unique @map(name: \"_id\")\n        field: String!\n      }\"\n    ");
    });
    (0, globals_1.it)('generates a simple model with no specified collection', function () {
        var mongo = index_1.connector.MongoDB(mongoParams);
        mongo.model('User', {
            id: index_1.g.id().unique().mapped('_id'),
            field: index_1.g.string()
        });
        index_1.g.datasource(mongo);
        (0, globals_1.expect)((0, utils_1.renderGraphQL)((0, index_1.config)({ schema: index_1.g }))).toMatchInlineSnapshot("\n      \"extend schema\n        @mongodb(\n          name: \"Test\"\n          apiKey: \"SOME_KEY\"\n          appId: \"test\"\n          dataSource: \"data\"\n          database: \"tables\"\n        )\n\n      type User @model(connector: \"Test\", collection: \"User\") {\n        id: ID! @unique @map(name: \"_id\")\n        field: String!\n      }\"\n    ");
    });
    (0, globals_1.it)('generates a decimal field', function () {
        var mongo = index_1.connector.MongoDB(mongoParams);
        mongo
            .model('User', {
            id: index_1.g.id().unique().mapped('_id'),
            field: index_1.g.decimal()
        })
            .collection('users');
        index_1.g.datasource(mongo);
        (0, globals_1.expect)((0, utils_1.renderGraphQL)((0, index_1.config)({ schema: index_1.g }))).toMatchInlineSnapshot("\n      \"extend schema\n        @mongodb(\n          name: \"Test\"\n          apiKey: \"SOME_KEY\"\n          appId: \"test\"\n          dataSource: \"data\"\n          database: \"tables\"\n        )\n\n      type User @model(connector: \"Test\", collection: \"users\") {\n        id: ID! @unique @map(name: \"_id\")\n        field: Decimal!\n      }\"\n    ");
    });
    (0, globals_1.it)('generates a bytes field', function () {
        var mongo = index_1.connector.MongoDB(mongoParams);
        mongo
            .model('User', {
            id: index_1.g.id().unique().mapped('_id'),
            field: index_1.g.bytes()
        })
            .collection('users');
        index_1.g.datasource(mongo);
        (0, globals_1.expect)((0, utils_1.renderGraphQL)((0, index_1.config)({ schema: index_1.g }))).toMatchInlineSnapshot("\n      \"extend schema\n        @mongodb(\n          name: \"Test\"\n          apiKey: \"SOME_KEY\"\n          appId: \"test\"\n          dataSource: \"data\"\n          database: \"tables\"\n        )\n\n      type User @model(connector: \"Test\", collection: \"users\") {\n        id: ID! @unique @map(name: \"_id\")\n        field: Bytes!\n      }\"\n    ");
    });
    (0, globals_1.it)('generates a bigint field', function () {
        var mongo = index_1.connector.MongoDB(mongoParams);
        mongo
            .model('User', {
            id: index_1.g.id().unique().mapped('_id'),
            field: index_1.g.bigint()
        })
            .collection('users');
        index_1.g.datasource(mongo);
        (0, globals_1.expect)((0, utils_1.renderGraphQL)((0, index_1.config)({ schema: index_1.g }))).toMatchInlineSnapshot("\n      \"extend schema\n        @mongodb(\n          name: \"Test\"\n          apiKey: \"SOME_KEY\"\n          appId: \"test\"\n          dataSource: \"data\"\n          database: \"tables\"\n        )\n\n      type User @model(connector: \"Test\", collection: \"users\") {\n        id: ID! @unique @map(name: \"_id\")\n        field: BigInt!\n      }\"\n    ");
    });
    (0, globals_1.it)('generates a model with auth', function () {
        var mongo = index_1.connector.MongoDB(mongoParams);
        mongo
            .model('User', {
            id: index_1.g.id().unique().mapped('_id'),
            field: index_1.g.string()
        })
            .collection('users')
            .auth(function (rules) {
            rules.private();
        });
        index_1.g.datasource(mongo);
        (0, globals_1.expect)((0, utils_1.renderGraphQL)((0, index_1.config)({ schema: index_1.g }))).toMatchInlineSnapshot("\n      \"extend schema\n        @mongodb(\n          name: \"Test\"\n          apiKey: \"SOME_KEY\"\n          appId: \"test\"\n          dataSource: \"data\"\n          database: \"tables\"\n        )\n\n      type User @model(connector: \"Test\", collection: \"users\") @auth(\n          rules: [\n            { allow: private }\n          ]) {\n        id: ID! @unique @map(name: \"_id\")\n        field: String!\n      }\"\n    ");
    });
    (0, globals_1.it)('generates a model with cache', function () {
        var mongo = index_1.connector.MongoDB(mongoParams);
        mongo
            .model('User', {
            id: index_1.g.id().unique().mapped('_id'),
            field: index_1.g.string()
        })
            .collection('users')
            .cache({ maxAge: 30 });
        index_1.g.datasource(mongo);
        (0, globals_1.expect)((0, utils_1.renderGraphQL)((0, index_1.config)({ schema: index_1.g }))).toMatchInlineSnapshot("\n      \"extend schema\n        @mongodb(\n          name: \"Test\"\n          apiKey: \"SOME_KEY\"\n          appId: \"test\"\n          dataSource: \"data\"\n          database: \"tables\"\n        )\n\n      type User @model(connector: \"Test\", collection: \"users\") @cache(maxAge: 30) {\n        id: ID! @unique @map(name: \"_id\")\n        field: String!\n      }\"\n    ");
    });
    (0, globals_1.it)('generates a two datasources with separate models', function () {
        var test = index_1.connector.MongoDB(mongoParams);
        var another = index_1.connector.MongoDB({
            name: 'Another',
            apiKey: 'OTHER_KEY',
            appId: 'foo',
            dataSource: 'bar',
            database: 'something'
        });
        test
            .model('User', {
            id: index_1.g.id().unique().mapped('_id')
        })
            .collection('users');
        another
            .model('Post', {
            id: index_1.g.id().unique().mapped('_id')
        })
            .collection('posts');
        index_1.g.datasource(test);
        index_1.g.datasource(another);
        (0, globals_1.expect)((0, utils_1.renderGraphQL)((0, index_1.config)({ schema: index_1.g }))).toMatchInlineSnapshot("\n      \"extend schema\n        @mongodb(\n          name: \"Test\"\n          apiKey: \"SOME_KEY\"\n          appId: \"test\"\n          dataSource: \"data\"\n          database: \"tables\"\n        )\n        @mongodb(\n          name: \"Another\"\n          apiKey: \"OTHER_KEY\"\n          appId: \"foo\"\n          dataSource: \"bar\"\n          database: \"something\"\n        )\n\n      type User @model(connector: \"Test\", collection: \"users\") {\n        id: ID! @unique @map(name: \"_id\")\n      }\n\n      type Post @model(connector: \"Another\", collection: \"posts\") {\n        id: ID! @unique @map(name: \"_id\")\n      }\"\n    ");
    });
});
